voluncheer/profiles/views/profile.py

```diff
@@ -47,16 +49,17 @@ def get_context_data(self, **kwargs):
             kwargs["opportunity_lists"] = opportunity_lists
         if user.is_volunteer:
             kwargs["volunteer"] = Volunteer.objects.get(pk=user)
-            badge_urls = []
-            badge_list = kwargs["volunteer"].badges.split(",")
-            for badge in badge_list:
-                try:
-                    badge_urls.append(Volunteer.BADGES[badge])
-                except KeyError:
-                    pass
-            kwargs["badge_urls"] = badge_urls
             volunteer_profile = Volunteer.objects.get(pk=user)
             kwargs["user_form"] = VolunteerChangeForm(instance=volunteer_profile)
+            if kwargs["volunteer"].badges is not None:
+                badge_urls = []
+                badge_list = kwargs["volunteer"].badges.split(",")
+                for badge in badge_list:
+                    try:
+                        badge_urls.append(Volunteer.BADGES[badge])
+                    except KeyError:
+                        pass
+                kwargs["badge_urls"] = badge_urls
```

Comment:
I rewrote the logic for passing badges to the template here, so most of it will be rewritten when we merge our branches.